codegen-server/src/test/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerBuilderGeneratorTest.kt

```diff
@@ -54,4 +123,195 @@ class ServerBuilderGeneratorTest {
             """,
         )
     }
+
+    @ParameterizedTest(name = "(#{index}) Generate default value. Required Trait: {1}, nulls: {2}, optionals: {3}")
+    @MethodSource("localParameters")
+    fun `default values are generated and builders respect default and overrides`(testConfig: TestConfig, setupFiles: (w: RustWriter, m: Model, s: RustSymbolProvider) -> Unit) {
+        val valuesMap = defaultValues.mapValues { if (testConfig.nullDefault) null else it.value }
+        val model = generateModel(testConfig.requiredTrait, valuesMap)
+        val symbolProvider = serverTestSymbolProvider(model)
+        val project = TestWorkspace.testProject(symbolProvider)
+        project.withModule(RustModule.public("model")) {
+            setupFiles(this, model, symbolProvider)
+
+            val rustValues = setupValuesForTest(testConfig.values)
+            val setters = if (testConfig.requiredTrait) structSetters(rustValues) else writable { }
```

Comment:
This seems odd. It's only setting the values on the builder when the members are `@required`. What we want is to not set them when the members are `@default`, and test that the built structure uses the modeled default values.

`@required` should have no bearing on our assertions / generated code. Note [the spec](https://smithy.io/2.0/spec/aggregate-types.html#structure-member-optionality) says both `@default` and `@required` should generate non-`Option`al structure fields, so we're only testing that case out of an abundance of caution.